README.md

```diff
@@ -3,152 +3,167 @@
 
-Inspiration was taken from https://github.com/joestelmach/laconic, but crel wont screw with your bad in-DOM event listeners, and is smaller, faster, etc...
+Inspiration was taken from [laconic](https://github.com/joestelmach/laconic), but Crel wont screw with your bad in-DOM event listeners, and it's smaller,
+faster, etc...
 
-# Usage #
-
-Signature:
-
-```javascript
-crel(tagName/dom element [, attributes, child1, child2, childN...])
-```
-
-where `childN` may be
-
- * a DOM element,
- * a string, which will be inserted as a `textNode`,
- * `null`, which will be ignored, or
- * an `Array` containing any of the above
+# Installing
 
 For browserify:
 
-```
+```bash
 npm i crel
 ```
 
 ```javascript
 var crel = require('crel');
 ```
 
+For AMD:
+
+```javascript
+require.config({paths: { crel: 'https://cdnjs.cloudflare.com/ajax/libs/crel/3.1.0/crel.min' }});
+require(['crel'], function(crel) {
+  // Your code
+});
+```
+
 For standard script tag style:
 
 ```html
-    <script src="crel.min.js"></script>
+<script src="crel.min.js"></script>
 ```
 
-To make some DOM:
+# Usage
 
-Example:
+Syntax:
+
+```javascript
+// Returns a DOM element
+crel(tagName / domElement, attributes, child1, child2, childN);
+```
+
+where `childN` may be:
+
+- a DOM element,
+- a string, which will be inserted as a `textNode`,
+- `null`, which will be ignored, or
+- an `Array` containing any of the above
+
+## Examples
 
 ```javascript
 var element = crel('div',
-    crel('h1', 'Crello World!'),
-    crel('p', 'This is crel'),
-    crel('input', {type: 'number'})
+  crel('h1', 'Crello World!'),
+  crel('p', 'This is crel'),
+  crel('input', { type: 'number' })
 );
 
 // Do something with 'element'
 ```
 
-You can create attributes with dashes, or reserved keywords, but using strings for the objects keys:
+You can add attributes that have dashes or reserved keywords in the name, by using strings for the objects keys:
 
 ```javascript
-crel('div', {'class':'thing', 'data-attribute':'majigger'});
+crel('div', { 'class': 'thing', 'data-attribute': 'majigger' });
 ```
 
-You can pass an already available element to crel, and it will be the target of the attributes/child elements
+You can define custom functionality for certain keys seen in the attributes
+object:
 
 ```javascript
-crel(document.body,
-    crel('h1', 'Page title')
-)
+crel.attrMap['on'] = function(element, value) {
+  for (var i in value) {
+    element.addEventListener(i, value[i]);
+  }
+};
+// Attaches an onClick event to the img element
+crel('img', { on: { click: function() { console.log('Clicked'); }}});
+```
+
+You can pass already available elements to Crel to modify their attributes / add child elements to them
+
+```javascript
+crel(document.body, crel('h1', 'Page title'));
 ```
 
 You can assign child elements to variables during creation:
 
 ```javascript
 var button,
-    wrapper = crel('div',
-        button = crel('button')
-    );
+  wrapper = crel('div', (button = crel('button')));

```

Comment:
I'd like to keep this multi-line, as it reads more easily.

Reply:
Done